stream.go

```diff
@@ -671,15 +671,25 @@ func (cs *clientStream) SendMsg(m interface{}) (err error) {
 	if !cs.desc.ClientStreams {
 		cs.sentLast = true
 	}
-	data, err := encode(cs.codec, m)
-	if err != nil {
-		return err
-	}
-	compData, err := compress(data, cs.cp, cs.comp)
-	if err != nil {
-		return err
+	preparedMsg, ok := m.(*PreparedMsg)
```

Comment:
This block is repeated 3 times. Can we factor this out into a helper function similar to this:
`func prepareMsg(m interface{}, codec baseCodec, cp Compressor, comp encoding.Compressor) (hdr, payload, data []byte)`

Reply:
The operations being put into the function could also through errors, so I suggest the function should be:
```go 
func prepareMsg(m interface{}, codec baseCodec, cp Compressor, comp encoding.Compressor) (err Error, hdr, payload, data []byte)
```